IdentityCore/src/MSIDAccount.m

```diff
@@ -78,7 +248,52 @@ - (NSString *)userIdentifier
     {
         return [NSString stringWithFormat:@"%@.%@", self.uid, self.utid];
     }
-    return nil;    
+    return self.uniqueUserId;
+}
+
+#pragma mark - Helpers
+
+- (NSString *)accountTypeString
+{
+    switch (self.accountType)
+    {
+        case MSIDAccountTypeAADV1:
+            return @"AAD";
+            
+        case MSIDAccountTypeMSA:
+            return @"MSA";
+            
+        case MSIDAccountTypeAADV2:
+            return @"MSSTS";
+            
+        default:
+            return @"Other";
+    }
+}
+
+static NSDictionary *accountTypes = nil;
+
+- (MSIDAccountType)accountTypeFromString:(NSString *)type
+{
+    if (!accountTypes)
+    {
+        accountTypes = @{@"AAD": @(MSIDAccountTypeAADV1),
+                         @"MSA": @(MSIDAccountTypeMSA),
+                         @"MSSTS": @(MSIDAccountTypeAADV2)};
+    }
+    
+    NSNumber *accountType = accountTypes[type];
+    return accountType ? [accountType integerValue] : MSIDAccountTypeOther;
+}
+
+#pragma mark - Update
+
+- (void)updateFieldsFromAccount:(MSIDAccount *)account
+{
+    NSMutableDictionary *allAdditionalFields = [NSMutableDictionary dictionary];
+    [allAdditionalFields addEntriesFromDictionary:account.additionalFields];
+    [allAdditionalFields addEntriesFromDictionary:_additionalFields];
```

Comment:
Current fields will override account.additionalFields. Is it intended behaviour?